### PR TITLE
docs: explain what the CI does

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

A highlight level overview what the CI does
Remove Jenkins badges and enable Buildkite badge

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

NA

## Related issues

NA

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+* [Microbenchmark](./microbenchmark.yml)
+* [Benchmarks](./benchmarks.yml)
+* Packaging - see below.
+
+## Scenarios
+
+* Tests should be triggered on branch, tag and PR basis.
+* Commits that are only affecting the docs files should not trigger any test or similar stages that are not required.
+* Builds do not get triggered automatically for Pull Requests from contributors that are not Elasticians when need to access to any GitHub Secrets.
```
[the remark this answers is not in the document]
Comment (v1v) [reply .github/workflows/README.md:19]: That sentence is not correct, thanks for spotting this out.
I think I might need to rephrase this to mention, GitHub secrets are not accessible from forked PRs, hence if secrets are needed to be accessed, then it's required to create a branch in the upstream

[the remark this answers is not in the document]
Comment (v1v) [reply .github/workflows/README.md]: https://github.com/elastic/apm-server/pull/11464/commits/6f943efcaf12fac8c9d1f51d1c5ac75616f27a23

**Review by amannocci — APPROVED** [review]
